### ISSUE TITLE
Code examples in API docs are not displayed with proper formatting in signature help

### ISSUE THREAD

**Issue by sameerajayasoma** [ballerina-platform/ballerina-lang#29547]
<img width="812" alt="Screen Shot 2021-03-23 at 4 04 48 PM" src="https://user-images.githubusercontent.com/2637328/112230096-8209ed80-8bf1-11eb-9d0e-e38ba8d97f90.png">

**Comment by nadeeshaan** [comment]
Following is the description extracted from the actual source

```
# Combines the members of an array using a combining function.
# The combining function takes the combined value so far and a member of the array,
# and returns a new combined value.
#
# + arr - the array
# + func - combining function
# + initial - initial value for the first argument of combining parameter `func`
# + return - result of combining the members of `arr` using `func`
#
# For example
# ```
# reduce([1, 2, 3], function (int total, int n) returns int { return total + n; }, 0)
# ```
# is the same as `sum(1, 2, 3)`.
```

When we set the documentation's  description for the symbol, we attach invalid content as shown bellow screen capture, which leads to invalid markdown.

<img width="1095" alt="Screenshot 2021-03-24 at 08 20 36" src="https://user-images.githubusercontent.com/1329674/112247459-597a0600-8c7a-11eb-8c7e-055229f71043.png">

**Comment by pubudu91** [comment]
Isn't this a parser issue then? In the semantic API, we simply get and set the documentation string we get from the internal symbols. Will check and verify.

**Comment by nadeeshaan** [comment]
> Isn't this a parser issue then? In the semantic API, we simply get and set the documentation string we get from the internal symbols. Will check and verify.

I hope so. As I understood, the internal symbol's documentation also contains this form.

**Comment by lochana-chathura** [comment]
Rendering issue fixed in PR  #30703.

Still for the above sample, the spec is not clear on what should happen when there is a code block or other documentation come after a parameter documentation block. https://github.com/ballerina-platform/ballerina-spec/issues/818

Until it is resolved current behavior seems to be okay. Hence closing this issue.
